services/navigation/path.go

```diff
@@ -2,23 +2,33 @@
 package navigation
 
 import (
+	"errors"
+
 	geo "github.com/kellydunn/golang-geo"
 	commonpb "go.viam.com/api/common/v1"
 	pb "go.viam.com/api/service/navigation/v1"
 )
 
+var (
+	errNilSlice = errors.New("cannot convert nil slice")
+	errNilPath  = errors.New("cannot convert nil path")
+)
+
 // Path describes a series of geo points the robot will travel through.
 type Path struct {
 	destinationWaypointID string
 	geoPoints             []*geo.Point
 }
 
 // NewPath constructs a Path from a slice of geo.Points and ID.
-func NewPath(id string, geoPoints []*geo.Point) *Path {
+func NewPath(id string, geoPoints []*geo.Point) (*Path, error) {
+	if geoPoints == nil {
```

Comment:
I think this should be checked with 
```
if len(goepoints) == 0 ...
```

b/c an empty slice could either be a nil pointer or an empty slice.

example:
```
// You can edit this code!
// Click here and start typing.
package main

import "fmt"

func main() {
	var a = []int{}
	fmt.Println(a)
	fmt.Println(a == nil)
	fmt.Println(len(a))
	a = nil
	fmt.Println(a)
	fmt.Println(a == nil)
	fmt.Println(len(a))
}
```

outputs:
```
 []
false
0
[]
true
0
```

If you are trying to prevent empty Path.geoPoints, you should check that with `len` not with an nil check.